uint/src/uint.rs

```diff
@@ -623,6 +692,25 @@ macro_rules! construct_uint {
 				arr[0]
 			}
 
+			/// Conversion to usize with overflow checking
+			///
+			/// # Panics
+			///
+			/// Panics if the number is larger than usize::max_value().
+			#[inline]
+			pub fn as_usize(&self) -> usize {
+				let &$name(ref arr) = self;
+				for i in 1..$n_words {
+					if arr[i] != 0 {
+						panic!("Integer overflow when casting U256")
```

Comment:
The panic description is a bit invalid (since it's `$name` not U256), but I'll log that as a separate issue.